File: Internshipviewpage/app/src/main/java/com/example/internshipviewpage/MainActivity.java
package com.example.internshipviewpage;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.widget.ViewPager2;

import android.os.Bundle;
import android.view.View;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    ViewPager2 viewPager2;
    ArrayList<viewpager_item> viewPagerItemArrayList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        viewPager2 = findViewById(R.id.viewpager);
        int[] images = {R.drawable.img1,R.drawable.img2,R.drawable.img3,R.drawable.img4,R.drawable.img5};
        String[] heading = {"Baked","Grilled","Dessert","Italian","Shakes"};
        String[] desc = {getString(R.string.img1_desc),
                getString(R.string.img2_desc),
                getString(R.string.img3_desc),
                getString(R.string.img4_desc)
                ,getString(R.string.img5_desc)};

        for (int i =0; i< images.length ; i++){

            viewpager_item viewPagerItem = new viewpager_item(images[i],heading[i],desc[i]);
            viewPagerItemArrayList.add(viewPagerItem);

        }

        vpAdapter vpAdapter = new vpAdapter(viewPagerItemArrayList);

        viewPager2.setAdapter(vpAdapter);

        viewPager2.setClipToPadding(false);

        viewPager2.setClipChildren(false);

        viewPager2.setOffscreenPageLimit(2);

        viewPager2.getChildAt(0).setOverScrollMode(View.OVER_SCROLL_NEVER);


        viewPagerItemArrayList = new ArrayList<>();
    }
}

File: View_withCompose/app/src/main/java/com/example/view_withcompose/MainActivity.kt
package com.example.view_withcompose

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.tooling.preview.Preview

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val composelayout=findViewById<ComposeView>(R.id.compose_layout)
        composelayout.setContent {
            sh(name = "jai Shree Ram")
        }
    }
}


@Composable



fun sh(name: String){
    Text(text = "Boliye $name")
}
